src/symbol/collision_feature.js

```diff
@@ -72,42 +72,60 @@ class CollisionFeature {
     _addLineCollisionBoxes(collisionBoxArray, line, anchor, segment, labelLength, boxSize, featureIndex, sourceLayerIndex, bucketIndex) {
         const step = boxSize / 2;
         const nBoxes = Math.floor(labelLength / step);
+        // We calculate line collision boxes out to 150% of what would normally be our
+        // max size, to allow collision detection to work on labels that expand as
+        // they move into the distance
+        const nPitchPaddingBoxes = Math.floor(nBoxes / 4);
```

Comment:
I was thinking it might be good to only have the extra boxes for pitched maps, but considering we're planning on taking a deep look at collision stuff soon it's probably not worthwhile to add that now. This looks good as is